Lusas_Adapter/CRUD/Create/Elements/Surface.cs

```diff
@@ -70,7 +74,39 @@ private IFSurface CreateSurface(Panel panel)
 
             IFSurface lusasSurface = d_LusasData.createSurfaceBy(edges.ToArray());
 
-            if (lusasSurface != null)
+            List<IFSurface> openings = new List<IFSurface>();
+
+            foreach (Opening opening in panel.Openings)
+               {
+                string openingID = GetAdapterId<string>(opening);
+
+                if (string.IsNullOrEmpty(openingID))
+                {
+                    Engine.Base.Compute.RecordWarning("Could not find the ids for at least one Opening, Opening not created.");
```

Comment:
```suggestion
                    Engine.Base.Compute.RecordError("Could not find the ids for at least one Opening, Opening not created.");
```
As this line of the code you will have the id of the surface, so you can include this in the error message to provide more information.